javalib/src/main/scala/java/util/concurrent/locks/ReentrantLock.scala

```diff
@@ -0,0 +1,48 @@
+package java.util.concurrent.locks
+
+import java.io.Serializable
+import java.lang.Thread
+import java.util.concurrent.TimeUnit
+
+class ReentrantLock(fair: Boolean) extends Lock with Serializable {
+
+  private var locked = 0
+
+  def this() = this(false)
+
+  def lock(): Unit = locked += 1
+
+  def lockInterruptibly(): Unit =
+    if (Thread.currentThread().isInterrupted()) throw new InterruptedException()
+    else locked += 1
+
+  def tryLock(): Boolean = {
+    locked += 1
+    true
+  }
+
+  def tryLock(time: Long, unit: TimeUnit): Boolean = {
+    if (Thread.currentThread().isInterrupted()) throw new InterruptedException()
+    tryLock()
+  }
+
+  def unlock(): Unit = {
+    if (locked <= 0) throw new IllegalMonitorStateException()
+    else locked -= 1
```

Comment:
What I meant by this was the coding style: if/else in statement position should have its branches on dedicated lines:
```scala
    if (locked <= 0)
      throw new IllegalMonitorStateException()
    else
      locked -= 1
```